.circleci/config.yml

```diff
@@ -0,0 +1,189 @@
+version: 2.1
+
+orbs:
+  aws-s3: circleci/aws-s3@1.0.11
+
+only_branches: &only_branches
+  filters:
+    branches:
+      ignore: master
+
+only_master: &only_master
+  filters:
+    branches:
+      only: master
+
+only_v_tags: &only_v_tags
+  filters:
+    tags:
+      only: /^v.*/
+    branches:
+      ignore: /.*/
+
+jobs:
+  build-ltops-and-test:
+    docker:
+      - image: circleci/golang:1.12
+    steps:
+      - checkout
+      - restore_cache:
+          key: dependencies-{{ checksum "go.sum" }}
+      - run:
+          name: Install dependencies
+          command: |
+            export PATH=$PATH:$GOPATH/bin
+            mkdir -p $GOPATH/bin
+            make install
+      - save_cache:
+          key: dependencies-{{ checksum "go.sum" }}
+          paths:
+            - $GOPATH/bin
+      - run:
+          name: Build package
+          command: make package
+      - run:
+          name: Test
+          command: |
+            set +x
+            ltops --help
+      - persist_to_workspace:
+          root: .
+          paths:
+            - dist
+
+  deploy-branches:
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - attach_workspace:
+          at: .
+      - aws-s3/sync:
+          from: dist/mattermost-load-test.tar.gz
+          to: s3://releases.mattermost.com/loadtest-pr/${CIRCLE_BRANCH}/
+          arguments: --cache-control no-cache
+
+  deploy-master:
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - attach_workspace:
+          at: .
+      - aws-s3/sync:
+          from: dist/mattermost-load-test.tar.gz
+          to: s3://releases.mattermost.com/mattermost-load-test/
+          arguments: --cache-control no-cache
+
+  deploy-tag:
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - attach_workspace:
+          at: .
+      - aws-s3/sync:
+          from: dist/mattermost-load-test.tar.gz
+          to: s3://releases.mattermost.com/loadtest/${CIRCLE_TAG}/
+          arguments: --cache-control no-cache
+
+  build-docker-branches:
```

Comment:
for this job you can make a shell test to check which branch is that and apply the correct labels if need
something like
```
if [[ $(echo "$CIRCLE_BRANCH" | grep -c "pull") -gt 0 ]]; then
    echo "PR."
    export DOCKER_TAG="${CIRCLE_SHA1}"
  else
    echo "Not a PR, so now do what I want."
    export DOCKER_TAG="${CIRCLE_BRANCH}"
  fi
```

also you can make the check to see if this is a tag, the env var is `CIRCLE_TAG`
you can get all the env vars you can use here: https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables

Reply:
please check again.

Reply:
@metanerd looks great thanks
now you can combine `master_build_and_deploy` and `tags_build_and_deploy` in one workflow

Reply:
I reworked the whole workflow again, because I learned about https://circleci.com/docs/2.0/reusing-config/#using-the-parameters-declaration 
Maybe it is more readable now? I hope so at least.